src/Extension.Tests/OutputEvaluationTests.cs

```diff
@@ -99,10 +99,10 @@ public void when_the_output_fails_any_rule_then_evaluation_fails()
             var ruleContext = new RuleContext();
       
             //act
-            var evaluator = new Evaluator();
+            var challenge = new Challenge(new EditableCode[] { });
 
-            evaluator.AddRule(new Rule(r => r.Fail()));
-            var evaluation = evaluator.EvaluateResult(ruleContext);
+            challenge.AddRule(new Rule(r => r.Fail()));
+            var evaluation = challenge.EvaluateResult(ruleContext);
```

Comment:
The "result" in `EvaluateResult` doesn't seem to refer to any of the relevant concepts in this API.

Reply:
Updated. For now, I've changed the name to `EvaluateByDefault` because to my understanding the challenge evaluation can be set by the teacher in the OnCodeSubmitted handler. And this function is kind of a default behavior (challenge evaluation is a conjunction of rule evaluations). FYI, @lytesfazion 

Somewhat related: this function shouldn't take in a RuleContext as a parameter since the RuleContext should be constructed within the function using information in the challenge. You can see this because the parameter is currently unused.

Reply:
I'm not sure I follow. Wouldn't that leave construction of the context object up to the teacher?

Reply:
My bad didnt explain clearly.The context objects should be constructed internally from within a challenge. Currently, this function takes in a rulecontext object as a parameter, which seems strange because "Where else would context object come from?"